### DISCUSSION TITLE
Web based build?

### DISCUSSION THREAD

**Discussion by bakaneko59** [Jyers/Marlin#764] [Ideas]
Have you considered providing a web based compile feature like Th3d does for their EZBOARD controller at https://ezfirmware.th3dstudio.com/ezconfig/

It would enable you to virtually eliminate the precompiled firmware and ultimately make it easier to support new features that you'd want to add.

Of course it would also require a custom website and extensive web programming to make it work, so it may not be worth your time. Maybe Th3d would consider providing their code for you to build upon. I just thought it seemed like a cool technique.

**Comment by t3chguy** [comment]
The programming wouldn't be the difficulty, paying for the compute for ad-hoc builds with basically an unlimited number of configuration options would probably exceed any free compute tiers available

**Comment by Jyers** [comment]
An interesting concept, but not really something I'm looking to sink a bunch of time into. I have created an auto compilation service for my patrons on patreon. It was really just intended as a perk for supporting me, but I suppose if someone really didn't want to or couldn't manage to compile their own that would be an option. It's not as user friendly as Th3d's versions, but all you need to do is make both configuration files and the rest is done for you..

**Reply by Cal-LF** [reply]
It looks like the three membership levels on your Patreon web site offer the same thing, auto compile service. 
FYI - The "Share" function for Facebook does not appear to work.

**Reply by Jyers** [reply]
They do, the other tiers are just there if you want to pledge more. I might think of other benefits for the higher tiers in the future, but for now they're all the same. As for the facebook thing, maybe I need to link my facebook? I'll see what I can do. I'm pretty new to Patreon.

**Comment by bakaneko59** [comment]
Yeah, I figured it might take a lot of work up front but might reduce your work in the long run. I was not familiar with your patron feature. I'll check it out.

**Comment by smooreace** [comment]
There is little work involved in pushing precompiled FW beyond what was
initially set up in Github. Its VERY unlikely that Th3D would be willing to
offer up their back end, but I could be willing to host the required site
if it came to that.

**Comment by tiagofreire-pt** [comment]
That web compiler would be awesome for some, but I prefer compiling myself the code. I'm curating a [highly customized configuration](https://github.com/tiagofreire-pt/Marlin/tree/Ender_3_V2_Extensible_UI/Marlin) for my usage and purposes, miles away from the available options on any pre-compiled or web compiler.

The effort pays a lot.

**Reply by Tuncay-Ayhan** [reply]
I'm doing the same thing. I'm staying in sync with Jacobs repo, but customizing my own stuff(not customized much. 

I'm even automating it now with Github Workflows. Took me 2 days to understand it, but it was worth the time and effort. Check my repo if you want. It's pretty much the same but with my own tweaks and only the builds I need. I even have beta's between builds now. So that's pretty awesome and easy to use. Forgot to mention. If i want to release files, I only have to push a "tag" in VSCode. Only that.... Then 5 minutes later a release is setup with all the files and everything done.

So that way I have the updated files and changes Jacob pushes through and then my config is bases on that. So I'll never have to miss it when he turns something on in the config, or turns it off. Splashes some bugs and add some features.

Life can be good.
